core/src/main/java/org/infinispan/commands/functional/TxReadOnlyManyCommand.java

```diff
@@ -107,6 +106,7 @@ public Object perform(InvocationContext ctx) throws Throwable {
             EntryView.ReadWriteEntryView rw = EntryViews.readWrite(entry);
             for (Mutation<K, V, ?> mutation : mutations) {
                ret = mutation.apply(rw);
+               entry.updatePreviousValue();
```

Comment:
I'm not sure why do you need to invoke this method in each iteration. can you elaborate? (same for the other places in TxDistributionInterceptor)